CT-1605 Do not show remove link cases where not allowed

stephenrichards opened the pull request:
For SAR cases, only managers are allowed to link cases; responders
are not allowed to link or unlink them.

This change removes the 'Remove link' link for responders on SAR cases.

misaka commented on app/helpers/cases_helper.rb:
```diff
@@ -37,6 +37,23 @@ def action_button_for(event)
+
+  def show_hide_message(kase)
+
+    (part1,part2) = kase.message_extract
```
Variable naming here is confusing. Not fatally so, it's a short method, but there is two part 2s here, etc. How about, simply:

```
content_tag(:span, part1, class: 'ellipse-preview')
  + content_tag(:span,'...', class:'ellipse-delimiter js-hidden')
  + content_tag(:span, part2,  class: 'ellipse-complete js-hidden' )
  + link_to 'Show more', '#', class: 'ellipse-button js-hidden'
```

misaka commented on app/helpers/cases_helper.rb:
```diff
@@ -40,17 +40,15 @@ def action_button_for(event)
+      content_tag(:span, preview_copy, class: 'ellipsis-preview')
+        + content_tag(:span,'...', class:'ellipsis-delimiter js-hidden')
+        + content_tag(:span, remaining_copy,  class: 'ellipsis-complete js-hidden' )
+        + link_to 'Show more', '#', class: 'ellipsis-button js-hidden'
```
Sorry, looks like it needs parens around this one.

aliuk2012 replied:
I did think the part1+part2.... seemed rather messy I prefer your suggestion of removing them

stephenrichards committed:
CT-1605 Do not show remove link cases where not allowed

For SAR cases, only managers are allowed to link cases; responders
are not allowed to link or unlink them.

This change removes the 'Remove link' link for responders on SAR cases.

beccapearce committed:
Merge branch 'master' into ct-1605-linking-sars
